services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_logs.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from asyncio import CancelledError, Task, create_task
+from contextlib import suppress
+from typing import Any, Callable, Coroutine, Dict, Optional, cast
+
+from fastapi import FastAPI
+
+from .rabbitmq import RabbitMQ
+from .utils import docker_client
+
+logger = logging.getLogger(__name__)
+
+
+async def _logs_fetcher_worker(
+    container_name: str, dispatch_log: Callable[..., Coroutine[Any, Any, None]]
+) -> None:
+    logger.info("Started log fetching for container %s", container_name)
+    async with docker_client() as docker:
+        container = await docker.containers.get(container_name)
+
+        logger.debug("Streaming logs from %s", container_name)
+        async for line in container.log(stdout=True, stderr=True, follow=True):
+            await dispatch_log(container_name=container_name, message=line)
+
+
+def _setup_logger(container_name: str) -> None:
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(
+        logging.Formatter("[%(levelname)-4s:%(name)-20s] %(message)s")
+    )
+    container_logger = logging.getLogger(container_name)
+    container_logger.addHandler(stream_handler)
+
+
+class BackgroundLogFetcher:
+    def __init__(self, app: FastAPI) -> None:
+        # requires rabbitmq to be in place
+        assert app.state.rabbitmq  # nosec
+
+        self._app: FastAPI = app
+
+        self._log_processor_tasks: Dict[str, Task[None]] = {}
+
+    @property
+    def rabbitmq(self) -> RabbitMQ:
+        return self._app.state.rabbitmq  # type: ignore
+
+    async def _dispatch_logs(self, container_name: str, message: str) -> None:
+        # logs from the containers will be logged at warning level to
+        # make sure they are not lost in production environments
+        # as these are very important to debug issues from users
+        logging.getLogger(container_name).warning(message)
```

Comment:
When you run in production mode you set the log level to `warning`. If the logs are set to something lower they will not appear.
That would be the name of the image. I guess that would be more informative that the name of the container. I'll give this a shot.

Reply:
Ok I think there are 2 different things here.
The logs of the service itself is always visible in Graylog, or Portainer. Therefore I do not see why we should add these into the dynamics sidecar as well. this is confusing. If I look into the dynamic sidecar I do not want to see the logs of the service as warnings. If I want to see the service logs, I will go in the service itself directly.
If the dynamic sidecar log level is set to WARNING, this is because we do not want to see INFO/DEBUG/TRACE logs. therefore I would not now artificially change that level.

Reply:
Ok, keeping the logs separated sounds good.
Will it from here.